packages/hint/docs/user-guide/development-flow-integration/travis-and-azure.md

```diff
@@ -0,0 +1,527 @@
+
+   ![Deployment slots menu button](images/slots-1.png)
+
+1. Select "Add Slot" in the blade that appears:
+
+   ![Add Slot button](images/slots-2.png)
+
+1. Choose a name (e.g.: "staging") and select a source to clone if
+   applicable (e.g.: "production"). This will copy all the
+   environment variables in the source such as `NODE_ENV=production`:
+
+   ![Menu to create a new slot on Azure](images/slots-3.png)
+
+1. Select the new slot in the menu and then "Deployment options"
+
+   ![Local Git Repository option](images/deployment-options-1.png)
+
+1. In the new menu, select "Local Git Repository"
+
+   ![Local Git Repository option](images/deployment-options-2.png)
+
+At this moment you might have to create some "deployment credentials".
+You can do so via the "Deployment credentials" option:
+
+![Deployment credentials option](images/deployment-credentials.png)
+
+## Create and configure a service principal
+
+This is an optional step, but recommended. You should create a service
+principal with only access to the bare minimum. That way you do not have
+to put your credentials anywhere (even if they are encrypted), and if
+something happens you can easily revoke access to it.
+
+1. To achieve this you first need to [install the Azure CLI][azure cli].
+   Then log in into your account using:
+
+   ```bash
+   az login
+   ```
+
+   A browser should open and give you access in the command line.
+
+   **Note**: If your account has access to multiple Azure subscriptions,
+make sure to chose the right one via
+`az account set --subscription "My Subscription"`.
+
+1. Then create a service principal:
+
+   ```bash
+   az ad sp create-for-rbac --name NAME --password PASSWORD --subscription "My Subscription"
+   ```
+
+   You should get an output similar to:
+
+   ```json
+   {
+      "appId": "b583eadf-7643-4b67-91a1-eb057d9fe24f",
+      "displayName": "NAME",
+      "name": "http://NAME",
+      "password": "PASSWORD",
+      "tenant": "22cfb846-dedd-4bc1-b664-83c738e456e0"
+   }
+    ```
+
+   You should write down this information, but the fields we will use later
+are:
+
+   * `name`
+   * `password`
+   * `tenant`
+
+1. By default, service principals are created with the `Contributor` role.
+   Unless you want this account to have access to anything, you should
+   remove it and scope it down. For this case, the `Website Contributor`
+   should be enough and you should give this role only for the Azure WebApp
+   you are using.
+
+   1. Go to the Azure Portal, select your Azure WebApp and click
+   on "Access control (IAM)".
+   1. Click on "+ Add".
+      ![Access control blade](images/access-control-1.png)
+   1. Search and select the service principal you created before.
+   1. Add the "Website Contributor" role and click Save.
+      ![Adding permissions to an account](images/access-control-2.png)
+
+## Add and configure `webhint` in your project
+
+First thing you have to do is add `webhint` to your project:
+
+```bash
+
```

Comment:
Remove empty line.

Reply:
There was an extra space in one of the closing ones. I'll update all the doc and remove all the enable/disable comments.